include/deal.II/hp/dof_handler.h

```diff
@@ -976,6 +985,17 @@ namespace hp
 
 }
 
+// The following is necessary for compilation under Visual Studio which is unable to correctly
+// distinguish between dealii::DoFHandler and dealii::hp::DoFHandler.
+// Plus it makes code in dof_handler.cc easier to read.
+// Requires C++11 support which is in Visual Studio 2013 and newer.
+#if _MSC_VER >= 1800
+template <int dim, int spacedim> using HpDoFHandler = ::dealii::hp::DoFHandler<dim, spacedim>;
+#else
+// When using older Visual Studio or a different compiler just fall back.
+#define HpDoFHandler DoFHandler
+#endif
```

Comment:
I think you want the `#else` branch to be `#elif defined(_MSC_VER)` to make sure it is still only visible to MS VC.